### PR TITLE
Various sys_net improvements & bugfixes

### PR DESCRIPTION
**Pull request by RipleyTom**

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 1]
```diff
@@ -1797,6 +1804,18 @@ error_code sys_net_bnet_getsockname(ppu_thread& ppu, s32 s, vm::ptr<sys_net_sock
 
 			return {};
 		}
+		#ifdef _WIN32
+		else
+		{
+			// windows doesn't support getsockname for sockets that are not bound
+			if (get_native_error() == 10022)
+			{
+				memset(&native_addr, 0, native_addrlen);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:1814]: Horrible hack, you can just save it in IDM instead of the operating system I think.

**Review by elad335 — CHANGES_REQUESTED** [review]

File: rpcs3/Emu/Cell/lv2/sys_net.cpp (hunk 1 above)
Comment (RipleyTom) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:1814]: I agree that windows is a horrible hack but I don't understand what you want to save in IDM?

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:1814]: sockname, dont rely on windows for it. unless it is expected to be empty for unbound sockets?

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 2]
```diff
@@ -2587,7 +2663,8 @@ error_code sys_net_bnet_setsockopt(ppu_thread& ppu, s32 s, s32 level, s32 optnam
 
 		if (optlen >= sizeof(s32))
 		{
-			std::memcpy(&native_int, optval_buf.data(), sizeof(s32));
+			native_int = *reinterpret_cast<be_t<s32> *>(optval_buf.data());
+			// std::memcpy(&native_int, optval_buf.data(), sizeof(s32));
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:2667]: remove this line.

File: rpcs3/Emu/Cell/lv2/sys_net.cpp (hunk 1 above)
Comment (RipleyTom) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:1814]: yes it's expected to be 0.0.0.0:0 if unbound

**Comment by JohnHolmesII** [comment]
Any thoughts on maybe consolidating the poll checks?

**Pull request merged**